cookbook/core/control_flow/dynamics.py

```diff
@@ -135,3 +135,68 @@ def wf(s1: str, s2: str) -> int:
 
 if __name__ == "__main__":
     print(wf(s1="Pear", s2="Earth"))
+
+
+# %%
+# Dynamic Workflows from Execution POV
+# ------------------------------------
+#
+# What Is a Dynamic Workflow?
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# A workflow whose directed acyclic graph (DAG) is computed at run-time is a :ref:`dynamic workflow <Dynamic Workflows>`. The tasks in a dynamic workflow are executed at runtime using dynamic inputs.
+#
+# Think of a dynamic workflow as a combination of a task and a workflow. It is used to dynamically decide the parameters of a workflow at runtime. It is both compiled and executed at run-time. You can define a dynamic workflow using the ``@dynamic`` decorator.
+#
+# Why Use Dynamic Workflows?
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# Flexibility
+# """""""""""
+#
+# Dynamic workflows simplify your pipelines, providing the flexibility to design workflows based on your project’s requirements, which can’t be achieved using static workflows.
+#
+# Lower Pressure on etcd
+# """""""""""""""""""""""
+#
+# The workflow CRD and the states associated with static workflows are stored in etcd, which is the Kubernetes database. This database stores Flyte workflow CRD as key-value pairs and keeps track of the status of each node’s execution.
+# A limitation of etcd is that the aggregate of the size of the workflow and the status of the nodes shouldn't exceed 2 MB.
```

Comment:
This 2MB can be inaccurate - EKS allows 1.5MB and GCP is 1.2MB I believe. Because these values are cluster dependent, and can change, I think we should rather state that etcd has a hard limit on data sizes.